### PR TITLE
Added the docs for the @dns-sensitive mocks

### LINKED ISSUE #6379: [PhpUnitBridge] Mock DNS functions
The PHPUnit bridge got support for mocking PHP's DNS related functions (see symfony/symfony#18181).

### PR DESCRIPTION
**Pull request by javiereguiluz**
This fixes #6379.

### CODE REVIEW

File: components/phpunit_bridge.rst [hunk 1]
```diff
@@ -203,14 +203,91 @@ And that's all!
...
+                'ipv6' => '::12',
+            ),
+        ),
+    ));
+
+Troubleshooting
+---------------
+
+The ``@group time-sensitive`` and ``@group dns-sensitive`` annotations work
+"by convention" and assumes that the namespace of the tested class can be
```
Comment (HeahDude) [inline components/phpunit_bridge.rst]: typo "assume" ?

File: components/phpunit_bridge.rst [hunk 2]
```diff
@@ -203,14 +203,91 @@ And that's all!
...
+obtained just by removing the ``\Tests\`` part from the test namespace. I.e.
+that if the your test case fully-qualified class name (FQCN) is
```
Comment (HeahDude) [inline components/phpunit_bridge.rst]: "if your test" ?

**Comment by stof** [comment]
@javiereguiluz given that the bridge has more and more features, wouldn't it be better to split its doc into multiple chapters instead, as done for other components ?

File: components/phpunit_bridge.rst [hunk 3]
```diff
@@ -203,14 +203,91 @@ And that's all!
     advances the internal clock the given number of seconds without actually
     waiting that time, so your test will execute 10 seconds faster.
 
-Troubleshooting
-~~~~~~~~~~~~~~~
+DNS-sensitive Tests
+-------------------
+
```
Comment (stof) [inline components/phpunit_bridge.rst]: missing ``versionadded``

**Comment by javiereguiluz** [comment]
I've changed everything reviewers suggested. I also agree with @stof and we should split this doc as soon as we try to add the explanation of a new feature.

**Comment by javiereguiluz** [comment]
Ready for the final review before merge. Thanks!

**Comment by xabbuh** [comment]
:+1:

**Comment by stof** [comment]
> we should split this doc as soon as we try to add the explanation of a new feature.

As this PR is adding explanations of new features, I would say it is time :smile:

**Comment by javiereguiluz** [comment]
@stof I was referring to new explanations after this feature :grin:

**Comment by xabbuh** [comment]
I am :+1: for merging this as is. If someone feels the need to split the component's documentation with this feature set, we can discuss that in a new PR, but don't block merging this one because of a new discussion.

File: components/phpunit_bridge.rst [hunk 4]
```diff
@@ -203,17 +203,97 @@ And that's all!
     advances the internal clock the given number of seconds without actually
     waiting that time, so your test will execute 10 seconds faster.
 
+DNS-sensitive Tests
+-------------------
+
+.. versionadded:: 2.7
```
Comment (WouterJ) [inline components/phpunit_bridge.rst]: `.. versionadded:: 3.1` ?

File: components/phpunit_bridge.rst [hunk 5]
```diff
@@ -203,17 +203,97 @@ And that's all!
...
+functions:
+
+* ``checkdnsrr()``
+* ``dns_check_record()``
+* ``getmxrr()``
+* ``dns_get_mx()``
+* ``gethostbyaddr()``
+* ``gethostbyname()``
+* ``gethostbynamel()``
+* ``dns_get_record()``
```
Comment (WouterJ) [inline components/phpunit_bridge.rst]: What about using PHP.net links here? ``:phpfunction:`dns_get_record` ``

File: components/phpunit_bridge.rst (hunk 4 above)
Comment (javiereguiluz) [inline components/phpunit_bridge.rst]: Fixed. Thanks!

File: components/phpunit_bridge.rst (hunk 5 above)
Comment (javiereguiluz) [inline components/phpunit_bridge.rst]: Done.

**Comment by WouterJ** [comment]
Thanks for the quick fixes, Javier!

**Pull request closed without merging**